(maint) Add command line options to ticketmatch

speedofdark opened the pull request.

scottsolowild commented on ticketmatch.rb:
```diff
@@ -206,8 +243,13 @@ def inspect
+end
+
+if jira_project_fixed_version == nil
+  jira_project_fixed_version = ask('Enter JIRA fix version: ') {|q| q.default = "#{jira_project_name} #{git_to_rev}"}
```
This line is a bit long. I'd wrap by converting `{`/`}` to `do`/`end` with newlines for readability.

mcdonaldseanp reviewed (approved):
Other than Scott's readability comment, LGTM

smcclellan reviewed (changes requested):
Can you squash all three of these commits into one? That way, in the commit history, there's no chance of the formatting fix or the README update being separated from the feature commit.

speedofdark committed:
(maint) add commandline arguments to ticketmatch

This adds the possibility of using command line arguments to specify what you
want ticketmatch to do. See the README for details. Note that the prompt-as-you-go
approach still works as before.

speedofdark commented:
Updated.

scottsolowild commented:
Just thought of this; might be good to add a `non-interactive` (or something like that) flag, so that if a pipeline misses a flag it won't stall indefinitely waiting for user input?

speedofdark commented:
At the moment, it prompts for anything you omit on the command line. Will do some research and come back to you.

speedofdark committed:
(maint) add --ci switch for running non-interactively for CI

If you specify --ci, all other command line arguments are required and will
abort with an exit status of 1 if not specified. This prevents forgetting an
argument causing hangs when running unattended.
